silence/silence.go

```diff
@@ -552,14 +605,22 @@ func QState(states ...SilenceState) QueryParam {
 
 // Query for silences based on the given query parameters.
 func (s *Silences) Query(params ...QueryParam) ([]*pb.Silence, error) {
+	start := time.Now()
+	defer func() {
+		s.metrics.queryDuration.Observe(float64(time.Since(start)) / float64(time.Second))
+	}()
+	s.metrics.queriesTotal.Inc()
+
 	q := &query{}
 	for _, p := range params {
 		if err := p(q); err != nil {
+			s.metrics.queryErrorsTotal.Inc()
```

Comment:
That's always pretty fragile when the return error is overwritten in a scope etc.
Went a different route that should solve it.